File: 성은/[PGS]짝지어제거하기.java
import java.util.Stack;

class Solution {

    public static int solution(String s) {
        Stack<Character> stack = new Stack<>();
        int idx = 0;

        if (s.length() == 0) {
            return 0;
        }

        stack.push(s.charAt(idx++));
        while (idx < s.length()) {
            char c = s.charAt(idx++);

            if (!stack.isEmpty() && stack.peek() == c) {
                stack.pop();
            } else {
                stack.push(c);
            }

        }

        return stack.isEmpty() ? 1 : 0;
    }
}

File: 유진/4월1주차/[PGS]짝지어 제거하기.js
function solution(s) {
  var answer = -1;
  let list = [];

  if (s.length % 2 != 0 || s == "") {
    answer = 0;
  } else {
    for (let i of s) {
      list.push(i);
      if (list[list.length - 1] == list[list.length - 2]) {
        list.pop();
        list.pop();
      }
    }
    if (list.length == 0) {
      answer = 1;
    } else {
      answer = 0;
    }
  }
  return answer;
}
